core/consensus/aura/src/lib.rs

```diff
@@ -496,7 +499,21 @@ fn check_header<B: Block, P: Pair>(
 		let public = expected_author;
 
 		if P::verify(&sig, &to_sign[..], public) {
-			Ok(CheckedHeader::Checked(header, digest_item))
+			match check_equivocation(client, slot_num, header.clone()) {
+				Ok(Some(equivocation_proof)) => {
+					// TODO: dispatch report here.
```

Comment:
Do we need to add a reporting module as part of this PR?

Reply:
A runtime module? I think we'll hold that off for another PR.